PhysicsTools/JetMCAlgos/plugins/HadronAndPartonSelector.cc

```diff
@@ -249,12 +251,20 @@ HadronAndPartonSelector::produce(edm::Event& iEvent, const edm::EventSetup& iSet
    }
 
    // select partons
-   if ( partonMode_!="Undefined" )
+   if ( partonMode_!="Undefined" ) {
      partonSelector_->run(particles,partons);
+     for(reco::GenParticleCollection::const_iterator it = particles->begin(); it != particles->end(); ++it)
+     {
+       if( !(it->status()==3 || (( partonMode_=="Pythia8" ) && (it->status()==23)))) continue;
```

Comment:
The particle for the physics definition are kept in prunedGenParticles.
Even if we change the standard definition in slimmed jets, I propose to still keep the particles for the algorithmic definition, that can still be calculated on top of MiniAOD level.